File: src/main/java/com/anod/appwatcher/backup/gdrive/SyncTask.java
package com.anod.appwatcher.backup.gdrive;

import android.content.Context;

import com.google.android.gms.common.api.GoogleApiClient;

/**
 * Created by alex on 7/29/14.
 */
public class SyncTask extends ApiClientAsyncTask<Boolean, Boolean, SyncTask.Result> {

    private final Listener mListener;

    public static class Result {
        public boolean status;
        public Exception ex;

        public Result(boolean status, Exception ex) {
            this.status = status;
            this.ex = ex;
        }
    }

    public interface Listener {
        public void onResult(Result result);
    }

    public SyncTask(Context context, Listener listener, GoogleApiClient client) {
        super(context, client);
        mListener = listener;
    }

    @Override
    protected Result doInBackgroundConnected(Boolean... params) {
        SyncConnectedWorker worker = new SyncConnectedWorker(mContext, getGoogleApiClient());
        try {
            worker.doSyncInBackground();
        } catch (Exception e) {
            AppLog.ex(e);
            return new Result(false, e);
        }

        return new Result(true, null);
    }

    @Override
    protected void onPostExecute(Result result) {
        mListener.onResult(result);
    }



}

File: src/main/java/com/anod/appwatcher/backup/gdrive/SyncTask.kt
package com.anod.appwatcher.backup.gdrive

import android.content.Context

import com.google.android.gms.common.api.GoogleApiClient

import info.anodsplace.android.log.AppLog

/**
 * @author alex
 * *
 * @date 7/29/14
 */
class SyncTask(private val context: Context, private val mListener: SyncTask.Listener, client: GoogleApiClient)
    : ApiClientAsyncTask(client) {

    interface Listener {
        fun onResult(result: Result)
    }

    override fun doInBackgroundConnected(): Result {
        val worker = SyncConnectedWorker(context, googleApiClient)
        try {
            worker.doSyncInBackground()
        } catch (e: Exception) {
            AppLog.e(e)
            return Result(false, e)
        }

        return Result(true, null)
    }

    override fun onPostExecute(result: Result) {
        mListener.onResult(result)
    }
}
